components/create-collective/CreateCollective_v2.js

```diff
@@ -0,0 +1,164 @@
+    createCollective: PropTypes.func,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = { collective: { type: 'COLLECTIVE' }, result: {} };
+    this.createCollective = this.createCollective.bind(this);
+    this.error = this.error.bind(this);
+    this.resetError = this.resetError.bind(this);
+    this.messages = defineMessages({
+      'host.apply.title': {
+        id: 'host.apply.title',
+        defaultMessage: 'Apply to create a new {hostname} collective',
+      },
+      'collective.create.title': {
+        id: 'collective.create.title',
+        defaultMessage: 'Create an Open Collective',
+      },
+      'collective.create.description': {
+        id: 'collective.create.description',
+        defaultMessage: 'The place for your community to collect money and share your finance in full transparency.',
+      },
+    });
+
+    this.host = props.host || {
+      type: 'COLLECTIVE',
+      settings: {
+        apply: {
+          title: this.props.intl.formatMessage(this.messages['collective.create.title']),
+          description: this.props.intl.formatMessage(this.messages['collective.create.description']),
+          categories: ['community', 'climate', 'opensource'],
+        },
+      },
+    };
+
+    this.next = props.host ? `/${props.host.slug}/apply` : '/create';
+  }
+
+  error(msg) {
+    this.setState({ result: { error: msg } });
+  }
+
+  resetError() {
+    this.error();
+  }
+
+  async createCollective(CollectiveInputType) {
+    if (!CollectiveInputType.tos) {
+      this.setState({
+        result: { error: 'Please accept the terms of service' },
+      });
+      return;
+    }
+    if (get(this.host, 'settings.tos') && !CollectiveInputType.hostTos) {
+      this.setState({
+        result: { error: 'Please accept the terms of fiscal sponsorship' },
+      });
+      return;
+    }
+    this.setState({ status: 'loading' });
+    CollectiveInputType.type = 'COLLECTIVE';
+    CollectiveInputType.HostCollectiveId = this.host.id;
+    if (CollectiveInputType.tags) {
+      // Meetup returns an array of tags, while the regular input stores a string
+      if (typeof CollectiveInputType.tags === 'string') {
+        CollectiveInputType.tags.split(',');
+      }
+
+      CollectiveInputType.tags =
+        Array.isArray(CollectiveInputType.tags) && CollectiveInputType.tags.length > 0
+          ? CollectiveInputType.tags.map(t => t.trim())
+          : null;
+    }
+    CollectiveInputType.tags = [...(CollectiveInputType.tags || []), ...(this.host.tags || [])] || [];
+    if (CollectiveInputType.category) {
+      CollectiveInputType.tags.push(CollectiveInputType.category);
+    }
+    CollectiveInputType.data = CollectiveInputType.data || {};
+    CollectiveInputType.data.members = CollectiveInputType.members;
+    CollectiveInputType.data.meetupSlug = CollectiveInputType.meetup;
+    delete CollectiveInputType.category;
+    delete CollectiveInputType.tos;
+    delete CollectiveInputType.hostTos;
```

Comment:
IMO we should decouple the UI state from the object we want to send to the backend.
In practice, I think we should move this to the validation in the form and make sure that the component that is calling this function is already sending the final version we're using in the request.

Reply:
So to clarify, you think we should be calling the mutation in the form component?

Reply:
I'm OK with calling the mutation from the parent component but I think everything that comes before the `try` statement should be in `CreateCollectiveForm`, either as validation (in the case of the first two conditionals) or inside the submit handler.
The idea behind this is keeping the logic behind `CollectiveInputType` isolated, and since this object is being created in `CreateCollectiveForm`, ideally, all its logic should live there.
WDYT?

Reply:
Update and now in component